core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3002,6 +2991,179 @@ private void optimiseOrderBy(QueryModel model, int topLevelOrderByMnemonic) {
+     * Advice should only be propagated if relevant.
+     * Cases:
+     *     ASOF JOIN
+     *         Propagate if the ordering is for the primary table only, and timestamp-first
+     *     OTHER JOINs
+     *         Propagate down primary table only.
+     *
+     * @param model The current query model
+     * @param jm The join model list for the current query model
+     * @param orderByMnemonic The advice 'strength'
+     * @param orderByDirectionAdvice The advice direction
+     */
+    private void pushDownOrderByAdviceToJoinModels(QueryModel model, ObjList<QueryModel> jm, int orderByMnemonic, IntList orderByDirectionAdvice) {
+        if (model == null) {
+            return;
+        }
+        // don't propagate though group by, sample by or distinct
+        if (model.getGroupBy().size() != 0
+                || model.getSampleBy() != null
+                || model.getSelectModelType() == QueryModel.SELECT_MODEL_DISTINCT) {
+            return;
+        }
+        // placeholder for prefix-stripped advice
+        ObjList<ExpressionNode> advice = null;
+        // Check if the orderByAdvice has names qualified by table names i.e 't1.ts' versus 'ts'
+        final boolean orderByAdviceHasDot = checkForDot(orderByAdvice);
+        // loop over the join models
+        for (int i = 0, n = jm.size(); i < n; i++) {
+            // get primary model
+            QueryModel jm1 = jm.getQuiet(i);
+            jm1 = jm1 != null ? jm1.getNestedModel() : null;
+            if (jm1 == null) {
+                return;
+            }
+            // get secondary model
+            QueryModel jm2 = jm1.getJoinModels().getQuiet(1);
+            // if order by advice has no table prefixes, we preserve original behaviour and pass it on.
+            if (!orderByAdviceHasDot) {
+                setAndCopyAdvice(jm1, orderByAdvice, orderByMnemonic, orderByDirectionAdvice);
+                optimiseOrderBy(jm1, orderByMnemonic);
+                return;
+            }
+            // if the order by advice is for more than one table, don't propagate it, as a sort will be needed anyway
+            if (i == 0 && !checkForConsistentPrefix(orderByAdvice)) {
+                return;
+            }
+            // if the orderByAdvice prefixes do not match the primary table name, don't propagate it
+            final CharSequence adviceToken = orderByAdvice.getQuick(0).token;
+            final int dotLoc = Chars.indexOf(adviceToken, '.');
+            if (!(Chars.equalsNc(jm1.getTableName(), adviceToken, 0, dotLoc)
+                    || (jm1.getAlias() != null && Chars.equals(jm1.getAlias().token, adviceToken, 0, dotLoc)))) {
+                optimiseOrderBy(jm1, orderByMnemonic);
+                return;
+            }
+            // order by advice is pushable, so now we copy it and strip the table prefix
+            if (advice == null) {
+                advice = duplicateAdviceAndTakeSuffix();
```

Comment:
No prefix does not mean the master table of the join:

```
  create table tab1 (
    id int,
    a int
  );
  
  create table tab2 (
    id int,
    b int
  );
  
  select a, b
  from tab1 join tab2 on tab1.id = tab2.id
  order by b 
```

There should be no order by advice to table `tab1` to pre-sort on column `b`.